File: src/main/java/org/example/collection/Coordinates.java
package org.example.collection;

public class Coordinates {
    private Float x; //Значение поля должно быть больше -645, Поле не может быть null
    private float y;
    public Coordinates(Float x, float y){
        this.x = x;
        this.y = y;
    }
    public Coordinates(){}

    public Float getX() {
        return x;
    }

    public float getY() {
        return y;
    }

    @Override
    public String toString() {
        return "Coordinates{" +
                "x=" + x +
                ", y=" + y +
                '}';
    }
}

File: src/main/kotlin/collection/Coordinates.kt
package org.chousik.collection

import java.util.*

class Coordinates(private var x: Float, private var y: Float) {


    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as Coordinates
        return y.compareTo(that.y) == 0 && Objects.equals(x, that.x)
    }

    override fun hashCode(): Int {
        return Objects.hash(x, y)
    }

    override fun toString(): String {
        return "Coordinates{" +
                "x=" + x +
                ", y=" + y +
                '}'
    }
}